tests/unit/test_pareto.py

```diff
@@ -145,10 +165,104 @@ def test_pareto_hypervolume_indicator_raises_for_reference_below_anti_ideal_poin
             [0.1, -0.6],
             0.37,
         ),
+        ([[2.0, 2.0, 0.0], [2.0, 0.0, 1.0], [3.0, 1.0, 0.0]], [4.0, 4.0, 4.0], 29.0),
     ],
 )
 def test_pareto_hypervolume_indicator(
     objectives: list[list[float]], reference: list[float], expected: float
 ) -> None:
     pareto = Pareto(tf.constant(objectives))
     npt.assert_allclose(pareto.hypervolume_indicator(tf.constant(reference)), expected)
+
+
+def test_pareto_divide_conquer_nd_two_dimension_case() -> None:
+    objectives = tf.constant(
+        [
+            [0.9575, 0.4218],
+            [0.9649, 0.9157],
+            [0.1576, 0.7922],
+            [0.9706, 0.9595],
+            [0.9572, 0.6557],
+            [0.4854, 0.0357],
+            [0.8003, 0.8491],
+            [0.1419, 0.9340],
```

Comment:
it will be easier to see what the test is testing with simpler numbers. With long numbers, there's more to read which makes test failures harder to debug, and also makes refactoring harder. A future developer might well wonder: are these numbers so detailed for a reason?

Reply:
not for a reason, just sampled it from unit square. I'll change it to shorter numbers